src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -266,7 +266,14 @@ export class TerminalService implements ITerminalService {
 				const persistentProcessId = instanceToDetach?.persistentProcessId;
 				if (persistentProcessId) {
 					await instanceToDetach.detachFromProcess();
-					await this._primaryOffProcessTerminalService?.acceptInstanceForAttachment(persistentProcessId);
+					const process = await this._primaryOffProcessTerminalService?.acceptDetachedInstance(e.requestId, persistentProcessId);
+					if (process) {
```

Comment:
I know we don't want to do this here, but i'm not sure where else it should happen